index.py

```diff
@@ -315,12 +319,13 @@ def get_size(self):
         return self.size
 
     # Insert with var byte encoding
-    def insert(self, doc_id, field, positions):
+    def insert(self, doc_id, field, positions, new_doc_id=True):
         next_id = self.size
         new_posting = Posting(next_id, doc_id, field, positions)
         new_posting.var_byte_encoding()
         self.postings.append(new_posting)
-        self.size += 1
+        if new_doc_id:
+            self.size += 1
```

Comment:
Let's rename `self.size` to `self.unique_docids` for clarity

Reply:
Though some code on the search side relies on the `size` attribute of PostingList, so those might have to be changed to `len(posting_list.postings)` before merging